### PR TITLE
feat: support AUTO_ID_CACHE

### PR DESCRIPTION
**Pull request by wd0517**
Refer to https://github.com/pingcap/django-tidb/issues/52

This PR plans to support `AUTO_ID_CACHE` in Django by declaring it in the model's meta option.

Note: `tidb_auto_id_cache` only affects table creation.

For example:

```
class Test(models.Model):
    a = models.TextField()

    class Meta:
        tidb_auto_id_cache = 1
```

This PR also removed `functions.py` because we have already reverted `DatabaseWrapper.vendor` to mysql, so there is no need to patch the as_tidb function anymore.

### CODE REVIEW

File: django_tidb/schema.py [hunk 1]
```diff
@@ -60,3 +60,10 @@ def add_field(self, model, field):
             self.execute(self._create_unique_sql(model, [field]))
         else:
             super().add_field(model, field)
+
+    def table_sql(self, model):
+        sql, params = super().table_sql(model)
+        tidb_auto_id_cache = getattr(model._meta, "tidb_auto_id_cache", None)
+        if tidb_auto_id_cache is not None:
+            sql += " AUTO_ID_CACHE %s" % tidb_auto_id_cache
```
Comment (dveeden) [inline django_tidb/schema.py:68]: What if  `tidb_auto_id_cache` isn't a number?

**Review by dveeden — COMMENTED** [review]
As `AUTO_ID_CACHE=1` makes things more MySQL compatible... should we consider making that the default? As this isn't the default in TiDB I don't think we should, but maybe others have different opinions on this?

File: django_tidb/schema.py (hunk 1 above)
Comment (wd0517) [reply django_tidb/schema.py:68]: The user will receive an error thrown by django-tidb or TiDB, Django does not provide a way to validate meta option, its built-in option can also result in error when got invalid value.

**Comment by zhangyangyu** [comment]
Do we need the `tidb` prefix? I think it's TiDB unique and already under `django-tidb` scope?

**Comment by zhangyangyu** [comment]
Seems after v6.4.0 make it default doesn't hurt anything.

**Comment by wd0517** [comment]
> Do we need the `tidb` prefix? I think it's TiDB unique and already under `django-tidb` scope?

I think has `tidb` prefix can help application developers understand that this attribute belongs to tidb.

**Comment by wd0517** [comment]
I have added a `TIDB_DEFAULT_AUTO_ID_CACHE = 1` which can be modified in `settings.py`.

**Comment by wd0517** [comment]
I revert the above change, because `AUTO_ID_CACHE=1` have bad performance before TiDB v6.4.0, if we set a default auto_id_cache in `django-tidb`, it will make things become complex.

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**